### PR TITLE
Update state machine

### PR DESCRIPTION
**Pull request by gitbuda**

### CODE REVIEW

**Comment by gitbuda** [comment]
Closes #28

File: src/connection/tests.rs [hunk 1]
```diff
@@ -2,54 +2,92 @@ use super::*;
...
+        Ok(records) => {
+            assert_eq!(connection.status, ConnectionStatus::Ready);
+            records
+        }
+        Err(err) => panic!("Fetching all failed: {}", err),
     }
 }
 
-fn execute_query(query: String) {
+pub fn initialize() -> Connection {
```
Comment (gitbuda) [inline src/connection/tests.rs:43]: Remove `pub`

File: src/connection/tests.rs [hunk 2]
```diff
@@ -95,530 +133,156 @@ fn from_connect_fetchone_panic_sslkey() {
...
-#[serial]
-fn from_connect_fetchone_address() {
-    initialize();
-    let connect_prms = ConnectParams {
-        address: Some(String::from("127.0.0.1")),
-        ..Default::default()
-    };
-    let connection = get_connection(connect_prms);
-    assert!(connection.lazy);
+fn execute_executing_panic() {
```
Comment (gitbuda) [inline src/connection/tests.rs:147]: Rename these `_panic` functions to `_error`.

File: src/connection/tests.rs [hunk 3]
```diff
@@ -95,530 +133,156 @@ fn from_connect_fetchone_panic_sslkey() {
...
-    };
-    let mut connection = get_connection(connect_prms);
-    let params = get_params("name".to_string(), "Alice".to_string());
-
-    let query = String::from("MATCH (n:User) WHERE n.name = $name RETURN n LIMIT 5");
-    match connection.execute(&query, Some(&params)) {
-        Ok(x) => x,
-        Err(err) => panic!("Query failed: {}", err),
-    };
+fn fetchall_closed_panic() {
```
Comment (gitbuda) [inline src/connection/tests.rs:163]: Wrong function name

**Comment by gitbuda** [comment]
@antaljanosbenjamin I think this is finally ready for review :smile:

**Review by antaljanosbenjamin — APPROVED** [review]
I created two issues (#34 and #35) regarding to our discussion yesterday. There are two minor comments from me, but it looks really nice. The status management become much readable.

**Pull request merged**